mitodl/mit-learn
# Change frontend to use resource_prices field instead of prices field

### What are the relevant tickets?
PR 2 of 5 to address https://github.com/mitodl/hq/issues/5134 and https://github.com/mitodl/hq/issues/4685

Dependent on PR 1  here: https://github.com/mitodl/mit-learn/pull/1736

### Description (What does it do?)
- Updates the front end to use the resource_prices field and display the correct currency symbol.
- Adds a new JS package [currency-symbol-map](https://www.npmjs.com/package/currency-symbol-map) to display the correct symbol for the currency.

### Screenshots (if appropriate):
If price is in Euros (EUR):
<img width="386" alt="Screenshot 2024-10-23 at 1 05 50 PM" src="https://github.com/user-attachments/assets/690d2057-cddf-4239-aa34-79e13ea29eec">


If price is in Swiss francs (CHF, no symbol):

<img width="880" alt="Screenshot 2024-10-23 at 1 03 54 PM" src="https://github.com/user-attachments/assets/d9b37536-d70d-4aad-ace2-155b4e2cb4b1">


In dollars (USD):
<img width="310" alt="Screenshot 2024-10-23 at 1 07 46 PM" src="https://github.com/user-attachments/assets/67f7d927-f6fb-4931-be3f-6dfadb7c302c">



### How can this be tested?
- If you don't already have some courses with prices, run these commands on the **main** branch:
  ```
  ./manage.py backpopulate_mitxonline_data
  ./manage.py backpopulate_xpro_data
  ./manage.py backpopulate_sloan_data
  ```
- Log in as an admin user to avoid caching and go to `http://open.odl.local:8062/search?platform=mitxonline&platform=xpro`, you will see a mix of courses that should all either be paid with certificate (xpro) or free with a paid certificate option (mitxonline).  If you click on a course card you will see the same price in the drawer.
- You can test the currency by running this code:
   ```python
   from learning_resources.models import LearningResource
   from learning_resources.utils import resource_upserted_actions
   resource = LearningResource.objects.get(title="Applying Machine Learning to Engineering and Science")
   resource.resource_prices.all().update(currency="EUR")
   resource_upserted_actions(resource, False)
   ```
- Now wait a minute then go to http://open.odl.local:8062/search?q=%22Applying+Machine+Learning%22  -  the price of the top result should show the euro symbol €1559 instead of $1559 on the course card and resource drawer.
- Also go to http://open.odl.local:8062/search?q=%22Driving+Strategic+Innovation%22 - the first result should have a price of "CHF 12500" (Swiss francs, which don't have a symbol).
- Check and uncheck the "Free" checkbox on the search page.  The results should change as expected.


### Additional Context
- There will be 3 subsequent PR's before https://github.com/mitodl/hq/issues/5134 is closed:
  - 3: Modify the `LearningResource.prices` and `LearningResourceRun.prices` fields to be the same as `resource_prices`, and update schema. (Note: this will cause index updates to fail until a reindex completes)
  - 4: Change frontend to use `prices` field again
  - 5: Remove `resource_prices` from models and schemas.
